react/OrderItems.tsx

```diff
@@ -570,6 +506,96 @@ export const OrderItemsProvider: FC = ({ children }) => {
     [enqueueTask, setOrderForm, updateItemsTask]
   )
 
+  /**
+   * Add the items to the order form.
+   *
+   * Returns if the items were added or not.
+   */
+  const addItem = useCallback(
+    (
+      items: Array<Partial<CatalogItem>>,
+      marketingData?: Partial<MarketingData>
+    ) => {
+      const { newItems, updatedItems } = items.reduce(
+        (acc, item) => {
+          const { newItems: newList, updatedItems: updateList } = acc
+          const existingItem = findExistingItem(item, orderFormItemsRef.current)
+
+          if (existingItem == null) {
+            newList.push(item)
+          } else {
+            updateList.push(item)
+          }
+
+          return acc
+        },
+        { newItems: [], updatedItems: [] } as Record<
+          string,
+          Array<Partial<CatalogItem>>
+        >
+      )
+
+      if (updatedItems.length) {
+        updatedItems.forEach((item) => {
+          const existingItem = findExistingItem(item, orderFormItemsRef.current)
+
+          updateQuantity({
+            ...item,
+            quantity: (item.quantity ?? 1) + existingItem!.quantity,
+          })
+        })
+      }
+
+      if (newItems.length === 0) {
+        return
+      }
+
+      const mutationInputItems = newItems.map(adjustForItemInput)
+      const orderFormItems = newItems.map(mapToOrderFormItem)
+
+      setOrderForm((prevOrderForm) => {
+        return {
+          ...prevOrderForm,
+          items: [...orderFormItemsRef.current, ...orderFormItems],
+          totalizers: orderFormItems.reduce(
+            (totalizers: Totalizer[], item: Item): Totalizer[] => {
+              return updateTotalizersAndValue({ totalizers, newItem: item })
+                .totalizers
+            },
+            (prevOrderForm.totalizers as Totalizer[]) ?? []
+          ),
```

Comment:
same as above

Reply:
This was already there. `prevOrderForm.totalizers` is a `Maybe<Totalizer>`, that's why the `as` is there.